### DISCUSSION TITLE
[Question]: How to configure a service which is not consumed by another service or controller

### DISCUSSION THREAD

**Discussion by dr194** [TypedProject/ts-express-decorators#2739] [Q&A] [answered]
I want to configure a service which is not injected by other service or by any controller. My usecase is that in the service, we are polling for an AWS SQS event. Since it is not invoked anywhere during the build, the service is not able to poll for events.

Few options which I have tried out:
1. ` componentsScan `: As per the Ts.ED documentation, it is [deprecated](https://tsed.io/docs/configuration.html#componentsscan-deprecated).
PS: Ts.ED documentation is contradicting for usage of ` componentsScan `. On [Configuration](https://tsed.io/docs/configuration.html#componentsscan-deprecated) doc, it says as deprecated but in [Service](https://tsed.io/docs/services.html#configuration) doc, its suggesting to make use of it. This needs to be corrected.

2. [@tsed/components-scan](https://www.npmjs.com/package/@tsed/components-scan): It works but lib warns not to be used in prod env (Ref: [Documentation](https://tsed.io/getting-started/migration-from-v6.html#keep-the-componentsscan))

Please help me with the correct way to inject a service in this scenario.

**Comment by Romakita** [comment, marked as answer]
The answer his in the doc:

https://tsed.io/docs/providers.html#services

Search es6 import

See you

**Comment by VictoriqueMoe** [comment]
it is also worth noting that if you use `barrelsby`, you can use that to create an `index.ts` file of your services and you can simply import that whole index file into your main entry point file. i do this here: https://github.com/waifuvault/WaifuVault/blob/main/src/Server.ts#L12

barrelsby does not work for ESM. so you can use this: https://github.com/waifuvault/WaifuVault/blob/main/buildScripts/fixbarrels.mjs
